pyro/ops/jit.py

```diff
@@ -9,6 +11,35 @@
 from pyro.util import ignore_jit_warnings, optional
 
 
+def _hash(value, allow_id):
+    try:
+        hash(value)
+        return value
+    except TypeError as e:
+        if isinstance(value, list):
+            return tuple(_hash(x, allow_id) for x in value)
+        elif isinstance(value, dict):
+            return tuple(sorted((_hash(x, allow_id), _hash(y, allow_id)) for x, y in value.items()))
+        elif isinstance(value, set):
+            return frozenset(_hash(x) for x in value)
```

Comment:
Should we have a default kwarg `allow_id=False` for this?

Reply:
no, it is an internal function